tests/base.py

```diff
@@ -0,0 +1,110 @@
+from importlib import import_module
+from inspect import Parameter, signature
+
+from sdk_commons.config import SDKConfig
+
+TEMPLATE = 'template'
+RIBBON = 'ribbon'
+OPYN = 'opyn'
+FRIKTION = 'friktion'
+# TODO: add FRIKTION
+VENUES = [TEMPLATE, RIBBON, OPYN]
+
+
+class TestsBase:
+    def setUp(self):
+        pass
+
+    @staticmethod
+    def import_module(*args: str) -> str:
+        return import_module(".".join(args))
+
+    @staticmethod
+    def import_class(*args: str) -> str:
+        module = import_module(".".join(args[0:-1]))
+        cls_name = args[-1]
+        assert hasattr(module, cls_name), f"{'.'.join(args)} not found"
+
+        c = getattr(module, cls_name)
+        assert c is not None
+
+        return c
+
+    def get_config_class(self, venue: str):
+        """
+        Scan the venue.config module to search
+        a class inheriting SDKConfig
+        """
+        module = self.import_module(venue, "config")
+        config_class = None
+
+        # Iterate the symbol table
+        for name, cls in module.__dict__.items():
+            # exclude all elements that are not classes
+            if not isinstance(cls, type):
+                continue
+            # exclude all classes that are from a different module
+            # i.e. exclude imported items
+            if module.__name__ not in cls.__module__:
+                continue
+
+            # exclude all classes not children of SDKConfig
+            if not issubclass(cls, SDKConfig):
+                continue
+
+            config_class = name
+            break
+
+        assert config_class is not None, f"Can't any class inheriting SDKConfig in {venue}.config"
+
+        return config_class
+
+    @staticmethod
+    def inspect_method_signature(method: callable, reference: callable):
```

Comment:
Maybe in the future we can experiment mypy to handle these validations 😀

Reply:
Totally totally totally agree! That would also super useful to validate return values, that we are currently unable to validate without executing the functions

Reply:
Wanna consider this?
https://github.com/tradeparadigm/sdks/runs/7494774772?check_suite_focus=true

Reply:
Nice! But it seems the mypy needs to wait for some work to Friktion?

Reply:
Not really, mypy is already verifying Friktion but nothing is changed on the code (only remove an unnecessary type ignore)